### ISSUE TITLE
Zernike moments - Getting wiered values after above 20 degree 

### ISSUE THREAD

**Issue by sanctusin** [luispedro/mahotas#96]
Trying to extract ZM from two images and compare them with opencv norm function to measure similarity, 
It works well for below 20 Degree but it doesn't capture detail information of images so for some images i am getting wrong similarity. 

I tried to increase Degree to 60 Degree and found out that same image from different image giving Negative number like -88835.23 which should be between 0 to 1 

can you please look into it and resolve the issue code below 
desc = mahotas.features.zernike_moments(imres, 200, degree=60, cm=200,200)
desc1 = mahotas.features.zernike_moments(imres1, 200, degree=60, cm=200,200)
score = cv2.norm(desc, desc1, cv2.NORM_L2)  

Thanks in advance

**Comment by luispedro** [comment]
Thanks. Would it be possible for you to upload the corresponding images?

It may be that numerically, it just blows up.

**Comment by sanctusin** [comment]
I am Soo Sorry for late answering. I was playing with SIFT and Bag-of-Visual-Words. 
will check and confirm if it is the numbers. also can the computations done in parallel to save time. CPU cores and memory are cheap but not time

**Comment by luispedro** [comment]
Thanks. It's been a while since I looked at this, but I think the numerics of these computations are such that they would have to be implemented much more carefully at higher moments for them to not blow up.

**Comment by jleechung** [comment]
Yeah, it seems more likely to happen when `n` is large and `l` is small:

https://github.com/luispedro/mahotas/blob/1c84c258c1e4167d4e6864abb3d399703894464d/mahotas/features/zernike.py#L97

which is when the factorials become large:

https://github.com/luispedro/mahotas/blob/1c84c258c1e4167d4e6864abb3d399703894464d/mahotas/features/_zernike.cpp#L60-L64

But even then for `n=50, l=0` these computations are way under the threshold of the double limit.

Do you have any intuition of where the blow up is occurring?
